### PR TITLE
[Go-gen] Add method for getting parent ID from a URL

### PR DESCRIPTION
**Pull request by c-harding**
Generate a method for extracting the parent ID from a URL, and output it whenever a service has structs.

Side effects: add a method for determining whether an AttributesRoot describes a struct

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
Cool cool cool

**Review by jaylees14 — APPROVED** [review]

File: src/e2e/resources/simple-temple-expected/simple-temple-test-user/util/util.go [hunk 1]
```diff
@@ -97,3 +97,13 @@ func ExtractAuthIDFromRequest(headers http.Header) (*Auth, error) {
 
 	return &Auth{uuid}, nil
 }
+
+// ExtractIDFromRequest extracts the parameter provided under parameter parent_id and converts it into a UUID
+func ExtractParentIDFromRequest(requestParams map[string]string) (uuid.UUID, error) {
+	id := requestParams["parent_id"]
+	if len(id) == 0 {
+		return uuid.Nil, errors.New("No Service ID provided")
```
Comment (willjvsmith) [inline src/e2e/resources/simple-temple-expected/simple-temple-test-user/util/util.go:105]: How come "No Service ID provided" as the error message?

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

File: src/e2e/resources/simple-temple-expected/simple-temple-test-user/util/util.go (hunk 1 above)
Comment (c-harding) [reply src/e2e/resources/simple-temple-expected/simple-temple-test-user/util/util.go:105]: This is the ID of the parent service. What would you suggest @smithwjv?

Comment (willjvsmith) [reply src/e2e/resources/simple-temple-expected/simple-temple-test-user/util/util.go:105]: Ah I read too quick, my mistake

Comment (willjvsmith) [reply src/e2e/resources/simple-temple-expected/simple-temple-test-user/util/util.go:105]: Thought it was fetching the struct ID

**Pull request merged**